src/stmt.cpp

```diff
@@ -1133,8 +1133,17 @@ void ForStmt::EmitCode(FunctionEmitContext *ctx) const {
     ctx->RestoreContinuedLanes();
     ctx->ClearBreakLanes();
 
-    if (step)
+    if (step) {
         step->EmitCode(ctx);
+        for (auto &inst : *bstep) {
+            // If the instruction is a user of phi node induction variable, set no wrap attributes.
+            // This enables the IndVarSimplify pass to detect the induction variable for simplification.
+            if (inst.isBinaryOp() || inst.isUnaryOp()) {
+                inst.setHasNoUnsignedWrap();
+                inst.setHasNoSignedWrap();
+            }
+        }
+    }
```

Comment:
Here's the patch for the sign extension hoisting issue. The loop needs to account for induction variable overflow during the step operation.